### PR TITLE
Improve performance of hot path in ShaderManager

### PR DESCRIPTION
**Pull request by peppy**
Enum.ToString() is quite an expensive operation, and was being called twice per drawable loaded.

### CODE REVIEW

File: osu-framework.sln.DotSettings [hunk 1]
```diff
@@ -181,6 +181,8 @@
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GC/@EntryIndexedValue">GC</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GL/@EntryIndexedValue">GL</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GLSL/@EntryIndexedValue">GLSL</s:String>
+  <s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=2D/@EntryIndexedValue">2D</s:String>
+  <s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=3D/@EntryIndexedValue">3D</s:String>
```
Comment (YaLTeR) [inline osu-framework.sln.DotSettings]: Somehow the indentation got ruined.

File: osu-framework.sln.DotSettings [hunk 2]
```diff
@@ -181,6 +181,8 @@
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GC/@EntryIndexedValue">GC</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GL/@EntryIndexedValue">GL</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=GLSL/@EntryIndexedValue">GLSL</s:String>
+	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=2D/@EntryIndexedValue">2D</s:String>
+	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=3D/@EntryIndexedValue">3D</s:String>
```
Comment (smoogipooo) [inline osu-framework.sln.DotSettings]: Are these rules still needed? Given that you're using TEXTURE_2/TEXTURE_3 below?

File: osu.Framework/Graphics/Containers/BufferedContainer.cs [hunk 3]
```diff
@@ -42,7 +42,7 @@ public class BufferedContainer<T> : Container<T> where T : Drawable
         private Vector2 blurSigma = Vector2.Zero;
 
         /// <summary>
-        /// Controls the amount of blurring in two orthogonal directions (X and Y if 
+        /// Controls the amount of blurring in two orthogonal directions (X and Y if
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Containers/BufferedContainer.cs]: Missing parens.

File: osu.Framework/Graphics/Shaders/ShaderManager.cs [hunk 4]
```diff
@@ -69,76 +69,53 @@ private ShaderPart createShaderPart(string name, ShaderType type, bool bypassCac
 
         public Shader Load(string vertex, string fragment, bool continuousCompilation = false)
         {
-            string name = $@"{vertex}/{fragment}";
+            string name = vertex + '/' + fragment;
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Shaders/ShaderManager.cs]: Was this profiled to a noticeable performance difference? If not I would prefer using string interpolation here.

File: osu-framework.sln.DotSettings (hunk 2 above)
Comment (peppy) [inline osu-framework.sln.DotSettings]: might as well have them

File: osu.Framework/Graphics/Shaders/ShaderManager.cs (hunk 4 above)
Comment (peppy) [inline osu.Framework/Graphics/Shaders/ShaderManager.cs]: was a measurable 5-10% overhead (due to this method being called ~100,000 times in the benchmark i was running)

File: osu.Framework/Graphics/Shaders/ShaderManager.cs [hunk 5]
```diff
@@ -69,76 +69,53 @@ private ShaderPart createShaderPart(string name, ShaderType type, bool bypassCac
...
+                {
+                    createShaderPart(vertex, ShaderType.VertexShader),
+                    createShaderPart(fragment, ShaderType.FragmentShader)
+                };
+
+                shader = new Shader(name, parts);
+
+                if (!shader.Loaded)
+                {
+                    StringBuilder logContents = new StringBuilder();
```
Comment (huoyaoyuan) [inline osu.Framework/Graphics/Shaders/ShaderManager.cs]: How about a conditional log method? This block may still exist even in Release build.

Comment (peppy) [inline osu.Framework/Graphics/Shaders/ShaderManager.cs]: For now I think this is fine. There's a lot of log output we will want to tidy up down the track.

Comment (Tom94) [inline osu.Framework/Graphics/Shaders/ShaderManager.cs]: The branch predictor will take care of it. Likely not a concern.

**Pull request merged**